src/plugins/discover/public/application/helpers/popularize_field.ts

```diff
@@ -7,13 +7,15 @@
  */
 
 import { IndexPattern, IndexPatternsContract } from '../../../../data/public';
+import { getServices } from '../../kibana_services';
 
 async function popularizeField(
   indexPattern: IndexPattern,
   fieldName: string,
   indexPatternsService: IndexPatternsContract
 ) {
-  if (!indexPattern.id) return;
+  const { capabilities } = getServices();
+  if (!indexPattern.id || !capabilities?.indexPatterns?.save) return;
```

Comment:
the use of `getServices` should be cleaned up in the future, it's a relict of our Kibana platform works. since it's partly done that way, could you move the check for the capability
to the code that is using `popularizeField`, like here? 

https://github.com/elastic/kibana/blob/328c36dedccc26229b34d9dad4b0b90714811234/src/plugins/discover/public/application/apps/main/components/doc_table/actions/columns.ts#L79-L83
This way there would be no cleanup necessary here

Reply:
I would prefer not to make the check in all 4 places we are calling that method, since it feels now the calling place needs to know how that method works internally to check for the right thing, but I am totally fine with saying: let's pass `capabilities` explicitly down and not use `getService()` for it. I'll change it to just pass `capabilities` into the `popularizeField` method as a parameter, if that's fine with you?

Reply:
for sure, this is the best solution!